### PR TITLE
fix mleibman/SlickGrid#855 (Memory leaks when cell contains jQuery controls

### PR DESCRIPTION
**Pull request by SimonBrangwin**

### CODE REVIEW

**Comment by 6pac** [comment]
A question, Simon, if you're still interested. 
In 

         $canvas[0].removeChild(cacheEntry.rowNode);
         $(cacheEntry.rowNode).empty(); // remove children from jQuery cache

I would imagine that the node will no longer exist by the time the second line executes, having been removed from the DOM. Is there something I'm missing ? 

I think we'd just use:

         $(cacheEntry.rowNode).remove();

Or - I'm thinking about removal of the element in an asynchronous fashion, similar to its original post-render instantiation - we'd use (this is a simplified version):

         $(cacheEntry.rowNode).detach();
         setTimeout(function(){ $(cacheEntry.rowNode).remove(); }, 100);

**Comment by 6pac** [comment]
(no need to mod your pull request, I'm formulating an API for my fork to queue the asynchronous cleanup of nodes. I'm just new to this kind of memory leak in jQuery)

**Comment by froodian** [comment]
It's worth noting that empty() or remove() both create the performance drag that was the subject of conversation in #855, so there's a trade-off here.

**Comment by froodian** [comment]
I've not moved our codebase over to your fork yet btw, but I am considering doing so, and I appreciate that you're keeping the project alive.

**Comment by 6pac** [comment]
Yes, the approach I've decided would be best, after much thought , is:
- global flag EnableCelljQueryCleanup and cellCleanupDelay
- nodesForCleanup array variable
- whenever DOM nodes are removed, check to see if they have been post processed, and if so, detach them, add them to the nodesForCleanup array and fire the cleanup function (if not post processed, they can just be deleted as usual)
- the cleanup function removes a single row's cells and re-calls itself after the given delay, just like the asyncpostrender master function

Re your codebase, I realise that once your own codebase has diverged, it can be difficult to apply the changes made to the original root. To that effect, I have offered discrete patches for most of the non trivial bug fixes. How have you found that? Are there any issue I should consider to make it easier for you?
Thanks for the feedback so far.

**Comment by froodian** [comment]
That cleanup approach sounds pretty good to me.  Discrete patches will probably help make this a little more manageable, but ultimately our copy of the code hasn't diverged that much from master, and now that you're maintaining your fork I should probably submit any changes we've made to you as patches as well - I just haven't been working on our project that uses slickgrid over the past few months, so it's just a prioritization issue for me to spend a day on this.

**Comment by 6pac** [comment]
A patch for this issue has been added to the 'alternative master' repo. An example page demonstrating cleanup has been added to the repo. Any testing, particularly regarding memory leaks, is greatly appreciated.
Repo: https://github.com/6pac/SlickGrid
Blog post: http://low-bandwidth.blogspot.com.au/2015/06/slickgrid-async-post-render-cleanup.html

I haven't tagged this commit as the next release yet, just waiting for this commit to settle.
